src/index.ts

```diff
@@ -30,12 +34,13 @@ const transact = (doc: Y.Doc | null, opts: Options, fn: () => void) => {
 export const bindProxyAndYMap = <T>(
   p: Record<string, T>,
   y: Y.Map<T>,
-  opts: Options = { transactionOrigin: null },
+  opts: Options = {},
 ) => {
+  const transactionOrigin = opts.transactionOrigin ?? (() => null);
   const pv2yvCache = new WeakMap<object, unknown>();
 
   const setPValueToY = (pv: T, k: string) => {
-    transact(y.doc, opts, () => {
+    transact(y.doc, transactionOrigin(), () => {
```

Comment:
Can this be like this?
```suggestion
    transact(y.doc, opts.transactionOrigin?.(), () => {
```
in other words, can it be `undefined` instead of `null`?

Reply:
Yes, that should work because the `origin` argument has `null` as a default value, so it will use `null` if we pass `undefined`: https://github.com/yjs/yjs/blob/645f05b0bb2435552582f587f5379c33adda6ffe/src/utils/Doc.js#L120

This allowed simplifying the code a bit.